lib/model-builder.js

```diff
@@ -168,6 +169,9 @@ ModelBuilder.prototype.define = function defineClass(className, properties, sett
     return ModelBaseClass.extend(className, properties, settings);
   }
 
+  // initializing base model inheritance rank if not set already
+  ModelBaseClass.__rank = ModelBaseClass.__rank || 1;
```

Comment:
I agree with @raymondfeng, I prefer calculating the model rank at bootstrapping time